res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -2067,157 +1754,109 @@ NumarkMixtrack3.OnPADLoopButtonChange = function(value, group, control) {
-                    ButtonNum + "]", "enabled");
-                deck.LEDs["fx" + ButtonNum].onOff(new_value ? ON : OFF);
-            }
-
+    var deck = NumarkMixtrack3.deckFromGroup(group);
+    var decknum = deck.decknum;
+    var effectNum = control - leds.fx1 + 1;
+    var effectGroup = "[EffectRack1_EffectUnit" + decknum + "_Effect" + effectNum + "]";
+
+    if (deck.PADMode && value === DOWN) {
+        if (deck.InstantFX.indexOf(effectNum) > -1) {
+            // remove effect from InstantFX and set LED to reflect whether or not effect is enabled
+            var index = deck.InstantFX.indexOf(effectNum);
+            deck.InstantFX.splice(index, 1);
+            deck.LEDs["fx" + effectNum].onOff(engine.getValue(effectGroup, "enabled") ? ON : OFF);
         } else {
+            // add new effect to InstantFX
+            deck.InstantFX.push(effectNum);
 
-            // we are adding the effect to the InstantFX list, or removing it
-            // tap + FX button enables/disables InstantFX, we check deck.TapDown to know if tap is pressed
-            //in order for FX LEDs to flas in sync, we need to loop thru the array to reset the LEDs
-            deck.InstantFX.push(ButtonNum);
-
-            // Get all LEDs to flash in sync
-            var arrayLength = deck.InstantFX.length;
-            for (i = 0; i < arrayLength; i++) {
-
-                ButtonNum = deck.InstantFX[i];
-                deck.LEDs["fx" + ButtonNum].flashOn(250, ON, 250);
+            // get all LEDs to flash in sync
+            for (var i = 0, n = deck.InstantFX.length; i < n; i++) {
+                effectNum = deck.InstantFX[i];
+                deck.LEDs["fx" + effectNum].flashOn(250, ON, 250);
             }
         }
-    } else if (value === DOWN && !deck.TapDown && !deck.PADMode) {
-        if (deck.shiftKey) {
-            // Select Effect 
-            if (FXmode === 1) { //3 FX in 4 FX Units
-                engine.setValue("[EffectRack1_EffectUnit" + decknum + "_Effect" + ButtonNum + "]",
-                    "next_effect", true); // Load FX, but not active
-                engine.setValue("[EffectRack1_EffectUnit" + decknum + "_Effect" + ButtonNum + "]",
-                    "enabled", false); // Load FX, but not active
-                engine.setValue("[EffectRack1_EffectUnit" + decknum + "]",
-                    "group_[Channel" + decknum + "]_enable", true); // An FX is loaded, activate Effect Unit
-            } else {
-                engine.setValue("[EffectRack1_EffectUnit" + ButtonNum + "]", "prev_chain", true);
-            }
+    } else if (deck.shiftKey && value === DOWN) {
+        // load next effect, make sure it is inactive, and make sure the unit is enabled
+        engine.setValue(effectGroup, "next_effect", true);
+        engine.setValue(effectGroup, "enabled", false);
+        engine.setValue("[EffectRack1_EffectUnit" + decknum + "]", "group_[Channel" + decknum + "]_enable", true);
```

Comment:
Currently switching decks also switches effect racks. This works with the Deere 2.1 when displaying 4 effect racks.

Personally I think 4 effect racks is overkill.. I think I'll try and have a toggle for the effect rack being applied to the current deck.. I'm worried that it may end up being confusing though.